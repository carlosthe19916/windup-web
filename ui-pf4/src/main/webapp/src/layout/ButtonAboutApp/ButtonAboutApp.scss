.about-app__component {
  .pf-c-about-modal-box__content {
    margin-top: 10px;
    margin-bottom: 20px;
  }
}
