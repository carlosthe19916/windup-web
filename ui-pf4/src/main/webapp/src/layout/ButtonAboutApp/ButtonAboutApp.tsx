import * as React from "react";
import {
  AboutModal,
  TextContent,
  TextList,
  TextListItem,
  Button,
  ButtonVariant,
} from "@patternfly/react-core";
import { HelpIcon } from "@patternfly/react-icons";
import brandImage from "../../img/mta-logo-header.svg";

export interface ButtonAboutProjectProps {}

interface State {
  isOpen: boolean;
}

export class ButtonAboutApp extends React.Component<
  ButtonAboutProjectProps,
  State
> {
  constructor(props: ButtonAboutProjectProps) {
    super(props);
    this.state = {
      isOpen: false,
    };
  }

  toggleButton = () => {
    this.setState(({ isOpen }) => ({
      isOpen: !isOpen,
    }));
  };

  render() {
    const { isOpen } = this.state;

    return (
      <>
        <Button
          id="aboutButton"
          aria-label="About button"
          variant={ButtonVariant.plain}
          onClick={this.toggleButton}
        >
          <HelpIcon />
        </Button>
        <AboutModal
          isOpen={isOpen}
          onClose={this.toggleButton}
          trademark="COPYRIGHT Â© 2020."
          brandImageSrc={brandImage}
          brandImageAlt="Logo"
          productName="Windup Web Console"
        >
          <TextContent>
            <TextList component="dl">
              <TextListItem component="dt">Source code</TextListItem>
              <TextListItem component="dd">
                <a href="https://github.com/windup/windup-web">Github</a>
              </TextListItem>
            </TextList>
          </TextContent>
        </AboutModal>
      </>
    );
  }
}
