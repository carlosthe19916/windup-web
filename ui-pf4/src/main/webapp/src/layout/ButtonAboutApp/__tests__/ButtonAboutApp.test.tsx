import React from "react";
import { shallow } from "enzyme";
import { ButtonAboutApp } from "../ButtonAboutApp";

it("Test snapshot", () => {
  const wrapper = shallow(<ButtonAboutApp />);
  expect(wrapper).toMatchSnapshot();
});

it("Test snapshot :: open modal", () => {
  const wrapper = shallow(<ButtonAboutApp />);
  wrapper.find("#aboutButton").simulate("click");
  expect(wrapper).toMatchSnapshot();
});
