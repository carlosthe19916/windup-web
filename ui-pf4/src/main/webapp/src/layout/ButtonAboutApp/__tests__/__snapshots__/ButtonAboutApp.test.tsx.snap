// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot :: open modal 1`] = `
<Fragment>
  <Button
    aria-label="About button"
    id="aboutButton"
    onClick={[Function]}
    variant="plain"
  >
    <HelpIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />
  </Button>
  <AboutModal
    appendTo={null}
    backgroundImageSrc=""
    brandImageAlt="Red Hat"
    brandImageSrc="red-hat-horizontal-reverse.svg"
    className=""
    isOpen={true}
    noAboutModalBoxContentContainer={false}
    onClose={[Function]}
    productName="Migration Toolkit for Applications"
    trademark=""
  >
    <TextContent
      className="pf-u-py-xl"
    >
      <h4>
        About
      </h4>
      <p>
        <a
          href="https://developers.redhat.com/products/mta/overview/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Migration Toolkit for Applications
        </a>
         
        allows application architects and developers to quickly decompile, analyze, assess and modernize large scale application portfolios and migrate them to Red Hat Middleware, cloud and containers.
      </p>
    </TextContent>
    <TextContent
      className="pf-u-py-xl"
    >
      <TextList
        component="dl"
      >
        <TextListItem
          component="dt"
        >
          Migration Toolkit for Applications Core
        </TextListItem>
        <TextListItem
          component="dd"
        >
          (loading)
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Migration Toolkit for Applications Web Console
        </TextListItem>
        <TextListItem
          component="dd"
        />
      </TextList>
    </TextContent>
    <TextContent>
      <h4>
        Links
      </h4>
      <List>
        <ListItem>
          <a
            href="https://developers.redhat.com/products/mta/overview/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Website
          </a>
        </ListItem>
        <ListItem>
          <a
            href="https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Documentation
          </a>
        </ListItem>
      </List>
    </TextContent>
  </AboutModal>
</Fragment>
`;

exports[`Test snapshot 1`] = `
<Fragment>
  <Button
    aria-label="About button"
    id="aboutButton"
    onClick={[Function]}
    variant="plain"
  >
    <HelpIcon
      color="currentColor"
      noVerticalAlign={false}
      size="sm"
    />
  </Button>
  <AboutModal
    appendTo={null}
    backgroundImageSrc=""
    brandImageAlt="Red Hat"
    brandImageSrc="red-hat-horizontal-reverse.svg"
    className=""
    isOpen={false}
    noAboutModalBoxContentContainer={false}
    onClose={[Function]}
    productName="Migration Toolkit for Applications"
    trademark=""
  >
    <TextContent
      className="pf-u-py-xl"
    >
      <h4>
        About
      </h4>
      <p>
        <a
          href="https://developers.redhat.com/products/mta/overview/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Migration Toolkit for Applications
        </a>
         
        allows application architects and developers to quickly decompile, analyze, assess and modernize large scale application portfolios and migrate them to Red Hat Middleware, cloud and containers.
      </p>
    </TextContent>
    <TextContent
      className="pf-u-py-xl"
    >
      <TextList
        component="dl"
      >
        <TextListItem
          component="dt"
        >
          Migration Toolkit for Applications Core
        </TextListItem>
        <TextListItem
          component="dd"
        >
          (loading)
        </TextListItem>
        <TextListItem
          component="dt"
        >
          Migration Toolkit for Applications Web Console
        </TextListItem>
        <TextListItem
          component="dd"
        />
      </TextList>
    </TextContent>
    <TextContent>
      <h4>
        Links
      </h4>
      <List>
        <ListItem>
          <a
            href="https://developers.redhat.com/products/mta/overview/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Website
          </a>
        </ListItem>
        <ListItem>
          <a
            href="https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Documentation
          </a>
        </ListItem>
      </List>
    </TextContent>
  </AboutModal>
</Fragment>
`;
